pkg/operator/ceph/config/probes_test.go

```diff
@@ -51,26 +52,24 @@ func configLivenessProbeHelper(t *testing.T, keyType cephv1.KeyType) {
 		},
 	}
 	container := v1.Container{LivenessProbe: p}
-	l := map[cephv1.KeyType]*cephv1.ProbeSpec{keyType: {Disabled: true}}
-	type args struct {
-		daemon      cephv1.KeyType
-		container   v1.Container
-		healthCheck cephv1.CephClusterHealthCheckSpec
+	store := cephv1.ObjectStoreSpec{}
+	fs := cephv1.CephFilesystem{}
+	healthCheck := cephv1.CephClusterHealthCheckSpec{}
+	livenessProbes := map[string]*cephv1.ProbeSpec{
+		"rgw": store.HealthCheck.LivenessProbe,
+		"mds": fs.Spec.MetadataServer.LivenessProbe,
+		"mon": healthCheck.LivenessProbe[keyType],
+		"osd": healthCheck.LivenessProbe[keyType],
+		"mgr": healthCheck.LivenessProbe[keyType],
 	}
-	tests := []struct {
-		name string
-		args args
-		want v1.Container
-	}{
-		{string(keyType) + "_probe-enabled", args{keyType, container, cephv1.CephClusterHealthCheckSpec{}}, container},
-		{string(keyType) + "_probe-disabled", args{keyType, container, cephv1.CephClusterHealthCheckSpec{LivenessProbe: l}}, v1.Container{}},
+	if got := ConfigureLivenessProbe(container, livenessProbes[string(keyType)]); !reflect.DeepEqual(got, container) {
```

Comment:
Passing in the expected output container to the input of this function doesn't properly validate its functionality. Whoever originally wrote this test probably should have called the above `container` `expectedContainer` or something similar. And I see they also made this mistake. I think it's easiest to use an empty container as input.

```suggestion
	if got := ConfigureLivenessProbe(v1.Container{}, livenessProbes[string(keyType)]); !reflect.DeepEqual(got, container) {
```